include/opendht/indexation/pht.h

```diff
@@ -296,26 +218,38 @@ class Pht {
     };
 
     /**
+     * Performs a step in the lookup operation. Each steps are performed
+     * asynchronously.
+     */
+    void lookupStep(Prefix k, std::shared_ptr<int> lo, std::shared_ptr<int> hi,
+            std::shared_ptr<std::vector<std::shared_ptr<Value>>> vals, LookupCallback cb,
+            DoneCallbackSimple done_cb, std::shared_ptr<unsigned> max_common_prefix_len,
+            int start = -1, bool all_values = false);
+
+    /**
      * Linearizes the key into a unidimensional key. A pht only takes
      * unidimensional key.
      *
      * @param Key  The initial key.
      *
-     * @return return The linearized key.
+     * @return the prefix of the linearized key.
      */
-    static Prefix linearize(Key k) {
-        if (k.size() != 1) { throw std::invalid_argument("PHT only supports unidimensional data."); }
-        return k.begin()->second;
+    virtual Prefix linearize(Key k) const {
+        if (not validKey(k)) { throw std::invalid_argument(INVALID_KEY); }
+        return Blob {k.begin()->second.begin(), k.begin()->second.begin() + keySpec_.begin()->second};
```

Comment:
Edit : After trying we do not get an error, however .. we cannot retrive the real Prefix since we get the prefix followed by 0.
We should use that we spoke about : a 1 and then fill with 0.

I will update this PR to add those modification since a need them for the issue #61